stories/__tests__/SQFormCheckboxGroup.stories.test.js

```diff
@@ -0,0 +1,191 @@
+  WithValidation: SQFormCheckboxGroupWithValidation
+} = composeStories(stories);
+
+const SHOPPING_LIST_OPTIONS = [
+  {label: 'Voltfruit', value: 'voltfruit'},
+  {label: 'Sunshroom', value: 'sunshroom'},
+  {label: 'Bokoblin Guts', value: 'bokoblin guts'},
+  {label: 'Lynel Hoof', value: 'lynel hoof'},
+  {label: 'Stealthfin Trout', value: 'stealthfin trout'}
+];
+
+describe('SQFormCheckboxGroup Tests', () => {
+  describe('Checkboxes Only', () => {
+    it('should render a label and a list of checkboxes', () => {
+      render(<SQFormCheckboxGroup size="auto" />);
+
+      const groupLabel = screen.getByText('Shopping List');
+      expect(groupLabel).toBeInTheDocument();
+
+      const checkboxes = screen.getAllByRole('checkbox');
+      expect(checkboxes.length).toBe(SHOPPING_LIST_OPTIONS.length);
+    });
+
+    it('should display as checked when initial values are set', () => {
+      const initialValues = {
+        shoppingList: [SHOPPING_LIST_OPTIONS[0].value]
+      };
+
+      render(<SQFormCheckboxGroup SQFormProps={{initialValues}} size="auto" />);
+
+      const checkboxLabel = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[0].label
+      );
+
+      expect(checkboxLabel).toBeChecked();
+
+      const nextCheckboxLabel = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[1].label
+      );
+      expect(nextCheckboxLabel).not.toBeChecked();
+    });
+
+    it('should show as checked when a checkbox is clicked', () => {
+      render(<SQFormCheckboxGroup size="auto" />);
+
+      const checkboxLabel = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[0].label
+      );
+      expect(checkboxLabel).not.toBeChecked();
+
+      userEvent.click(checkboxLabel);
+      expect(checkboxLabel).toBeChecked();
+    });
+
+    it('should uncheck when checkbox is checked and user clicks', () => {
+      render(<SQFormCheckboxGroup size="auto" />);
+
+      const checkboxLabel = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[1].label
+      );
+      expect(checkboxLabel).not.toBeChecked();
+
+      userEvent.click(checkboxLabel);
+      expect(checkboxLabel).toBeChecked();
+
+      userEvent.click(checkboxLabel);
+      expect(checkboxLabel).not.toBeChecked();
+    });
+
+    it('should render as required when isRequired is true', () => {
+      render(<SQFormCheckboxGroup size="auto" isRequired />);
+
+      const required = screen.getByText('Required');
+      expect(required).toBeInTheDocument();
+    });
+
+    it('should render an extra checkbox when adding select all capability', () => {
+      render(<SQFormCheckboxGroup size="auto" shouldUseSelectAll />);
+
+      const selectAllCheckbox = screen.getByLabelText('All');
+      expect(selectAllCheckbox).toBeInTheDocument();
+      expect(selectAllCheckbox).not.toBeChecked();
+    });
+
+    it('should check all checkboxes when select all checkbox checked', () => {
+      render(<SQFormCheckboxGroup size="auto" shouldUseSelectAll />);
+
```

Comment:
Here's another one where getting all the checkboxes then filtering it by 'checked' would well. 
You could:

- get all the checkboxes
- expect the checkboxes array filtered by `checked` to have length `0`
- click the select all checkbox
- expect the checkboxes array filtered by `checked` to be the shopping list length + 1 (for the select all checkbox)

Reply:
I have tried a few different ways to do this, but I can't find any attribute or way to actually check if the checkbox is checked. It worked in the above test, so I don't know why it won't work for this one.

Reply:
Ah, I missed the checked attribute in the filter in my previous comment. Updated that - it should have been `.filter(checkbox => checkbox.checked)`